R/extractData2.R

```diff
@@ -243,7 +250,7 @@ char2fac <- function(dat, labels, vars, convertMiss, ordered = FALSE) {
       partially_labeled <- partially_labeled[partially_labeled != i]
       if(all(fac_meta$value == seq(nrow(fac_meta)))) unordered_facs <- unordered_facs[unordered_facs != i]
 
-      dat[, i] <- factor(dat[, i], levels = fac_meta$valLabel, ordered = ordered)
+      dat[, i] <- factor(ori_dat[, i], levels = fac_meta$value, labels = fac_meta$valLabel, ordered = ordered)
```

Comment:
It is true that `fac_meta` and `dat` could be used to recreate the original values in `ori_dat`. However, this would basically mean reversing what has been done before (transforming the variable from numeric to character)... Not an ideal solution either and maybe even more confusing.

All in all, this is a hotfix and a general rework of extractData2() is aimed at in #101. But I agree with you, this is currently very suboptimal.